modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts

```diff
@@ -0,0 +1,288 @@
+      .get('/api/v1/client/constants')
+      .reply(200, { ttl: 3600, constants });
+
+    const bitgo = TestBitGo.decorate(BitGo, { env: 'mock' });
+    bitgo.initializeTestVars();
+
+    const baseCoin = bitgo.coin(coinName);
+
+    bgUrl = common.Environments[bitgo.getEnv()].uri;
+
+    const walletData = {
+      id: '5b34252f1bf349930e34020a00000000',
+      coin: 'gteth',
+      coinSpecific: {},
+    };
+    wallet = new Wallet(bitgo, baseCoin, walletData);
+    tssUtils = new EcdsaUtils(bitgo, baseCoin, wallet);
+  });
+
+  describe('TSS key chains:', async function() {
+    it('should generate TSS key chains', async function () {
+      const nockedBitGoKeychain = await nockBitgoKeychain({
+        coin: coinName,
+        userKeyShare,
+        backupKeyShare,
+        userGpgKey,
+      });
+      const nockedUserKeychain = await nockUserKeychain({ coin: coinName });
+      await nockBackupKeychain({ coin: coinName });
+
+      const bitgoKeychain = await tssUtils.createBitgoKeychain(userGpgKey, userKeyShare, backupKeyShare);
+      const userKeychain = await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase');
+      const backupKeychain = await tssUtils.createBackupKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase');
+
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
+      userKeychain.should.deepEqual(nockedUserKeychain);
+
+      // unencrypted `prv` property should exist on backup keychain
+      const keyChainPrv = JSON.parse(backupKeychain.prv ?? '');
+      _.isEqual(keyChainPrv.pShare, backupKeyShare.pShare).should.be.true();
+      _.isEqual(keyChainPrv.bitgoNShare, bitgoKeyShare.nShares[2]).should.be.true();
+      _.isEqual(keyChainPrv.userNShare, userKeyShare.nShares[2]).should.be.true();
+      should.exist(backupKeychain.encryptedPrv);
+    });
+
+    it('should generate TSS key chains with optional params', async function () {
+      const enterprise = 'enterprise';
+      const nockedBitGoKeychain = await nockBitgoKeychain({
+        coin: coinName,
+        userKeyShare,
+        backupKeyShare,
+        userGpgKey,
+      });
+      const nockedUserKeychain = await nockUserKeychain({ coin: coinName });
+      await nockBackupKeychain({ coin: coinName });
+
+      const bitgoKeychain = await tssUtils.createBitgoKeychain(userGpgKey, userKeyShare, backupKeyShare, enterprise);
+      const userKeychain = await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase',
+        'originalPasscodeEncryptionCode');
+      const backupKeychain = await tssUtils.createBackupKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase');
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
```

Comment:
Not sure we need to test it against nockKeychains, we are implementing the same logic there right?, we can just check if the shares are being actually generated with optional parameters and just check all the relevant properties are there correctly may be?